bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableOptionsFactory.java

```diff
@@ -396,6 +410,24 @@ public static BigtableOptions fromConfiguration(final Configuration configuratio
     if (tracingCookie != null) {
       bigtableOptionsBuilder.setTracingCookie(tracingCookie);
     }
+
+    final String s = configuration.get(BIGTABLE_TEST_DATA_IP_REGEX);
+    if (s != null) {
+      final ChannelConfigurator oldConfigurator = bigtableOptionsBuilder.getChannelConfigurator();
+      bigtableOptionsBuilder.setChannelConfigurator(new ChannelConfigurator() {
+        @Override
+        public ManagedChannelBuilder configureChannel(ManagedChannelBuilder builder, String host) {
+          if (oldConfigurator != null) {
+            builder = oldConfigurator.configureChannel(builder, host);
+          }
+          if (!host.contains("admin")) {
+            builder = builder.intercept(new IpVerificationInterceptor(Pattern.compile(s)));
```

Comment:
I'm a little worried about adding this code into the prod path when it's really just meant for testing. Is there another way we could do this in the test itself?

Reply:
The only other way is to inject classes into the Configuration, which quickly becomes a rube goldberg machine:

```
Class configuratorCls = config.getClass("google.bigtable.connection.configurator.class", ChannelConfigurator.class, NoopClass.class);
ChannelConfigurator instance = configuratorCls.newInstance();
instance.configureChannel(builder, host);

class DirectPathIpv4InterceptorConfigurator implements ChannelConfigurator {
// filtering for ipv4
}

class DirectPathIpv6InterceptorConfigurator implements ChannelConfigurator {
// filtering for ipv6
}
class DirectPathAnyInterceptorConfigurator implements ChannelConfigurator {
// filtering for ipv4 or ipv6
}
class CfeInterceptorConfigurator implements ChannelConfigurator {
// ! Direct path filtering
}
```
Which makes this a lot more difficult to deal with. Veneer already has this test, so once we transition to veneer, this can go away. It will be removed in 2.0 beta along with bigtable-client-core